text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+---
+feature name: aws-construct-library-module-lifecycle
+start date: 2020-01-29
+rfc pr: (leave this empty)
+related issue: 0107
+---
+
+# Summary
+
+The purpose of this document is to outline the process used to prioritize and mature modules in the AWS-CDK Construct Library. It outlines how we graduate modules from CloudFormation-only resources (L1) to modules that contain high-level resource abstractions (L2) and stabilize them for GA release.
+
+# Motivation
+
+We want to be transparent about the processes the AWS CDK team use, and seek community input to make them better.
+
+# Stages of an AWS-CDK Construct Library Module
+
+The sections below outline the development stages for modules in the AWS-CDK Construct Library and the criteria to graduate the module to the next level of maturity. There are 4 stages of development for an AWS-CDK Construct Library module: 
+
+* Stage 0 - CFN resources (L1) only 
+* Stage 1 - Experimental
+* Stage 2 - Developer Preview
+* Stage 3 - GA 
+
+## Stage 0 - CFN resources (L1) only
+
+All construct library modules start in stage 0 when they are auto-generated from the CloudFormation resource specification. The goal of stage 0 is to make new CloudFormation resources/properties available to CDK customers as soon as possible, and create tracking artifacts that allow us to capture the data required to make prioritization decisions to add L2 resources in the future. 
+
+* **Stage 0 activities:**
```

Comment:
When we initially reviewed the public roadmap plan, most people were of the opinion that we should not list things on the roadmap if we are not working on them. This applies to a majority of the CFN-only modules in stage 0. 

I've been thinking about having a "wish list" or "to do" column on the roadmap that lists all the CFN-only modules, so that customers have a convenient list to make their +1 votes. I think this can be important if customer +1s is a primary mechanism for prioritizing construct library work.